### PR TITLE
add custom http requrest(#3359)

### PR DESCRIPTION
**Pull request by baiyaaaaa**
Please makes sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer relative issues for you PR.
#3359

### CODE REVIEW

File: examples/docs/en-US/upload.md [hunk 1]
```diff
@@ -364,6 +364,7 @@ thumbnail-mode | whether thumbnail is displayed | boolean | — | false
 file-list | default uploaded files, i.e: [{name: 'food.jpeg', url: 'https://fuss10.elemecdn.com/3/63/4e7f3a15429bfda99bce42a18cdd1jpeg.jpeg?imageMogr2/thumbnail/360x360/format/webp/quality/100'}] | array | — | []
 list-type | type of fileList | string | text/picture/picture-card | text |
 auto-upload | whether to auto upload file | boolean | — | true |
+http-request | override for the default xhr behavior which allow you to implement your own upload-file's request | function | — | — |
```
Comment (Leopoldthecoder) [inline examples/docs/en-US/upload.md]: override default xhr behavior, allowing you to implement your own upload-file's request

**Comment by baiyaaaaa** [comment]
ddddddone

**Comment by yuchonghua** [comment]
@baiyaaaaa 能否支持一下Promise呢 https://github.com/ElemeFE/element/pull/3363

**Comment by yuchonghua** [comment]
@QingWei-Li  这个支持大概什么时候能发布到npm呢？最近有新版本吗？最近希望能快点有个新版本，因为这个比较急

**Pull request merged**